OnBaseDocsApi/Controllers/OnBaseDocsController.cs

```diff
@@ -55,7 +58,164 @@ public IHttpActionResult GetFile(int id)
+                        if (docAttr != null)
+                            return ConflictResult("More than one document attributes was included.");
+
+                        docAttr = JObject
+                            .Parse(content.ReadAsStringAsync().Result)
+                            .ToObject<DocumentPostAttributes>();
+                    }
+                }
+            }
+            catch (Exception ex)
+            {
+                return InternalErrorResult(ex.Message);
+            }
+
+            if ((docStream == null) || string.IsNullOrWhiteSpace(docExt))
+                return ConflictResult("The required parameter document content is missing.");
+            if (docAttr == null)
+                return ConflictResult("The required parameter document attributes are missing.");
+            if (string.IsNullOrWhiteSpace(docAttr.IndexKey))
+                return BadRequestResult("The required parameter 'IndexKey' is empty.");
+            if (string.IsNullOrWhiteSpace(docAttr.DocumentType))
+                return BadRequestResult("The required parameter 'DocumentType' is empty.");
+
+            return TryHandleRequest(app =>
+            {
+                // The document is moved to staging when there are no keywords included
+                // so that we can kick off a re-index to generate the autofill keywords.
+                bool toStaging = (docAttr.Keywords == null) || !docAttr.Keywords.Any();
+
+                var createAttr = new DocumentCreateAttributes
+                {
+                    DocumentType = toStaging ? Global.Config.StagingDocType : docAttr.DocumentType,
+                    Comment = docAttr.Comment,
+                    IndexKey = docAttr.IndexKey,
+                    Keywords = docAttr.Keywords,
+                    Ext = docExt,
+                    Stream = docStream,
+                    ToStaging = toStaging,
+                };
+                var result = CreateDocument(app, createAttr, out var doc);
+                if (result != null)
+                    return result;
+
+                if (toStaging)
+                {
+                    // We must get the document ID from this thread and not the task thread since
+                    // the document could go out of scope before the task starts.
+                    var docId = doc.ID;
+                    Task.Run(() =>
+                    {
+                        MoveDocumentToWorkflow(docId, docAttr.DocumentType);
+                    });
+                }
+
+                return DocumentResult(doc);
+            });
+        }
+
+        IHttpActionResult CreateDocument(Application app, DocumentCreateAttributes attr, out Document doc)
+        {
+            doc = null;
+
+            var docType = app.Core.DocumentTypes.Find(attr.DocumentType);
+            if (docType == null)
+                return InternalErrorResult($"The DocumentType '{attr.DocumentType}' could not be found.");
```

Comment:
I was able to pass in arbitrary strings like `"testType"` and didn't get this error. Is this working correctly?